src/cli.rs

```diff
@@ -0,0 +1,647 @@
+	println!("\t\t num_peers: {}", peer_manager.get_peer_node_ids().len());
+	println!("\t}},");
+}
+
+fn list_peers(peer_manager: Arc<PeerManager>) {
+	println!("\t{{");
+	for pubkey in peer_manager.get_peer_node_ids() {
+		println!("\t\t pubkey: {}", pubkey);
+	}
+	println!("\t}},");
+}
+
+fn list_channels(channel_manager: Arc<ChannelManager>) {
+	print!("[");
+	for chan_info in channel_manager.list_channels() {
+		println!("");
+		println!("\t{{");
+		println!("\t\tchannel_id: {},", hex_utils::hex_str(&chan_info.channel_id[..]));
+		println!(
+			"\t\tpeer_pubkey: {},",
+			hex_utils::hex_str(&chan_info.remote_network_id.serialize())
+		);
+		let mut pending_channel = false;
+		match chan_info.short_channel_id {
+			Some(id) => println!("\t\tshort_channel_id: {},", id),
+			None => {
+				pending_channel = true;
+			}
+		}
+		println!("\t\tpending_open: {},", pending_channel);
+		println!("\t\tchannel_value_satoshis: {},", chan_info.channel_value_satoshis);
+		println!("\t\tchannel_can_send_payments: {},", chan_info.is_live);
+		println!("\t}},");
+	}
+	println!("]");
+}
+
+fn list_payments(inbound_payments: PaymentInfoStorage, outbound_payments: PaymentInfoStorage) {
+	let inbound = inbound_payments.lock().unwrap();
+	let outbound = outbound_payments.lock().unwrap();
+	print!("[");
+	for (payment_hash, payment_info) in inbound.deref() {
+		println!("");
+		println!("\t{{");
+		println!("\t\tamount_millisatoshis: {},", payment_info.amt_msat);
+		println!("\t\tpayment_hash: {},", hex_utils::hex_str(&payment_hash.0));
+		println!("\t\thtlc_direction: inbound,");
+		println!(
+			"\t\thtlc_status: {},",
+			match payment_info.status {
+				HTLCStatus::Pending => "pending",
+				HTLCStatus::Succeeded => "succeeded",
+				HTLCStatus::Failed => "failed",
+			}
+		);
+
+		println!("\t}},");
+	}
+
+	for (payment_hash, payment_info) in outbound.deref() {
+		println!("");
+		println!("\t{{");
+		println!("\t\tamount_millisatoshis: {},", payment_info.amt_msat);
+		println!("\t\tpayment_hash: {},", hex_utils::hex_str(&payment_hash.0));
+		println!("\t\thtlc_direction: outbound,");
+		println!(
+			"\t\thtlc_status: {},",
+			match payment_info.status {
+				HTLCStatus::Pending => "pending",
+				HTLCStatus::Succeeded => "succeeded",
+				HTLCStatus::Failed => "failed",
+			}
+		);
+
+		println!("\t}},");
+	}
+	println!("]");
+}
+
+pub(crate) fn connect_peer_if_necessary(
+	pubkey: PublicKey, peer_addr: SocketAddr, peer_manager: Arc<PeerManager>,
+	event_notifier: mpsc::Sender<()>,
+) -> Result<(), ()> {
+	for node_pubkey in peer_manager.get_peer_node_ids() {
+		if node_pubkey == pubkey {
+			return Ok(());
+		}
+	}
+	match TcpStream::connect_timeout(&peer_addr, Duration::from_secs(10)) {
+		Ok(stream) => {
+			let peer_mgr = peer_manager.clone();
+			let event_ntfns = event_notifier.clone();
+			tokio::spawn(async move {
+				lightning_net_tokio::setup_outbound(peer_mgr, event_ntfns, pubkey, stream).await;
```

Comment:
@TheBlueMatt any idea why outbound connections to RL nodes isn't working? the node ID never shows up in `get_peer_node_ids`. Outbound connections to lnd work.